src/test/java/be/seeseemelk/mockbukkit/block/state/BlockStateMockTest.java

```diff
@@ -94,4 +94,16 @@ public void testUpdateForce()
 		assertTrue(chest.update(true));
 		assertTrue(block.getState() instanceof Chest);
 	}
+
+	@Test
+	public void testEquals() {
+		Block block = new BlockMock(Material.STONE, new Location(null, 0, 64, 0));
+
+		assertEquals(block.getState(), block.getState());
```

Comment:
What's the meaning of this test?

Do you want to test if the equals work fine with two references to the same instance? In that case, you may have to specify it at the test method name or explain it in a comment or somewhere.

Reply:
Ah, I didn't realise that calling getState multiple times in MockBukkit returns the same object.
The intent of the test was to check if different instances of the state are tested as the same, but yes, I'll clarify it as your comment.

Reply:
I took a look into the code of the getState() and will end with a call to `BlockStateMock#getSnapshot`

```java
/**
	 * This returns a copy of this {@link BlockStateMock}. Inheritents of this class should override this method!
	 * 
	 * @return A snapshot of this {@link BlockStateMock}.
	 */
	@NotNull
	public BlockState getSnapshot()
	{
		return new BlockStateMock(this);
	}
```

And that will create a new instance for each call, so your method may make sense.

Reply:
OK, thanks for the clarification.
I'll comment the test and make that other change in a bit.

Reply:
I would add some explanatory comments to avoid a future confusion.

Something like this one:
https://github.com/seeseemelk/MockBukkit/blob/6672eca92ec41e58121c84ef2ed92c9a98ec3301/src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java#L245-L250